TableBookingAPI/TableBooking.Logic/Repositories/GenericRepository.cs

```diff
@@ -35,10 +35,23 @@ public async Task Delete(object id)
             _objectSet.Remove(objectToDelete);
         }
 
-        public async virtual Task Update(T entity)
+        public async Task Update(T entity)
         {
-            await Update(entity);
-        }
+            var existingEntity = await _objectSet.FindAsync(GetKeyValues(entity));
+
+            if (existingEntity != null)
+            {
+                _context.Entry(existingEntity).State = EntityState.Detached;
+            }
 
```

Comment:
In my opinion logic for checking if something exist should be separate from update(but create and delete as well). 
You might have case when you need to check if something exist then you do update operation. With this code you will call database 2 times same request